MintegralAdapter/src/main/java/com/chartboost/mediation/mintegraladapter/MintegralAdapter.kt

```diff
@@ -802,68 +688,12 @@ class MintegralAdapter : PartnerAdapter {
             ad.playVideoMute(if (mute) MBridgeConstans.REWARD_VIDEO_PLAY_MUTE else MBridgeConstans.REWARD_VIDEO_PLAY_NOT_MUTE)
             ad.setRewardVideoListener(
-                object : RewardVideoListener {
-                    fun resumeOnce(result: Result<PartnerAd>) {
-                        if (continuation.isActive) {
-                            continuation.resume(result)
-                        }
-                    }
-
-                    override fun onVideoLoadSuccess(p0: MBridgeIds?) {
-                        PartnerLogController.log(LOAD_SUCCEEDED)
-                        resumeOnce(Result.success(PartnerAd(ad = ad, details = emptyMap(), request = request)))
-                    }
-
-                    override fun onLoadSuccess(p0: MBridgeIds?) {
-                    }
-
-                    override fun onVideoLoadFail(
-                        p0: MBridgeIds?,
-                        error: String?,
-                    ) {
-                        PartnerLogController.log(LOAD_FAILED, "$error")
-                        resumeOnce(Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN)))
-                    }
-
-                    override fun onAdShow(p0: MBridgeIds?) {
-                        onShowSuccess()
-                    }
-
-                    override fun onAdClose(
-                        p0: MBridgeIds?,
-                        rewardInfo: RewardInfo?,
-                    ) {
-                        PartnerLogController.log(DID_REWARD)
-                        listener.onPartnerAdRewarded(
-                            PartnerAd(ad = ad, details = emptyMap(), request = request),
-                        )
-                        PartnerLogController.log(DID_DISMISS)
-                        listener.onPartnerAdDismissed(
-                            PartnerAd(ad = ad, details = emptyMap(), request = request),
-                            null,
-                        )
-                    }
-
-                    override fun onShowFail(
-                        p0: MBridgeIds?,
-                        p1: String?,
-                    ) {
-                        onShowFailure()
-                    }
-
-                    override fun onVideoAdClicked(p0: MBridgeIds?) {
-                        PartnerLogController.log(DID_CLICK)
-                        listener.onPartnerAdClicked(
-                            PartnerAd(ad = ad, details = emptyMap(), request = request),
-                        )
-                    }
-
-                    override fun onVideoComplete(p0: MBridgeIds?) {
-                    }
-
-                    override fun onEndcardShow(p0: MBridgeIds?) {
-                    }
-                },
+                RewardedAdLoadCallback(
+                    listener,
+                    request,
+                    PartnerAd(ad = ad, details = emptyMap(), request = request),
```

Comment:
pretty sure the linter is going to put that on separate lines but I guess it's fine.